vars/deployClusterPipeline.groovy

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/groovy
+
+def call(String jobName, String stableNamespace, String stableImageTag, Map<String,String> tagOverrides) {
+  pipeline {
+    agent any
+
+    parameters {
+      string(name: "BUILD_ID")
+      string(name: "JOB_SPEC")
+      string(name: "REPO_OWNER")
+      string(name: "REPO_NAME")
+      string(name: "PULL_REFS")
+      string(name: "PULL_BASE_REF")
+      string(name: "PULL_BASE_SHA")
+      string(name: "PULL_NUMBER")
+      string(name: "PULL_PULL_SHA")
+    }
+
+    stages {
+      stage ("Configure Variables") {
+        steps {
+          script {
+            this._buildName = buildName(this)
+            this._jobId = "${jobName}-${this._buildName}-${env.BUILD_NUMBER}"
+          }
+        }
+      }
+      stage ("Ensure logging components exist") { steps {
+          ensureLoggingComponents(this)
+        }
+      }
+      stage("Extract Config Vars") {
+        steps {
+          script {
+            openshift.withCluster() {
+              def cm = openshift.selector("cm/${this._buildName}").object()
+              this._imageBase = cm.data["image-base"]
+              this._rpmRepo = cm.data["rpm-repo"]
+              this._kubeconfig = cm.data["kubeconfig"]
```

Comment:
How does this exist before the deploy?

Reply:
This will exist if the deploy has already happened. It's a way to make it idempotent

Reply:
Not sure we want that right now -- different e2e suites against the same PR should not share a cluster unless we can confirm that they are non-destructive